src/lib/services/__tests__/emissionsCalculator.test.ts

```diff
@@ -53,6 +55,57 @@ const withExtendedRanges: IntensityResponseData = new Array(25).fill(null).map((
 })
 
 describe('EmissionsCalculator', function () {
+  const mockCarbon = new MockAgent().get(`https://api.carbonintensity.org.uk`)
+
+  describe('fetchEmissions', function () {
+    const startDateBelow30Min = new Date('2023-01-01T07:00:00.000Z')
+    const endDateBelow30Min = new Date('2023-01-01T07:10:00.000Z')
+
+    this.beforeEach(() => {
+      setGlobalDispatcher(mockCarbon)
```

Comment:
You should make sure you have an `afterEach` here to clean up and reset the dispatcher. Otherwise you're poluting the test environment for all other unit tests

Reply:
Mmm... I've included 
```js
dispatcher.destroy()
agent.close()
```
Wonder if you could confirm that this is the correct termination, but this is what I found so far in docs
- https://undici.nodejs.org/#/docs/api/Agent
- https://undici.nodejs.org/#/docs/api/Dispatcher

Reply:
I'm not sure exactly but if you have a look in the integration test helper you'll see how we did this before?